app/src/common/shared/com/igalia/wolvic/utils/BitmapCache.java

```diff
@@ -75,11 +101,13 @@ void initDiskCache() {
     }
 
     public void addBitmap(@NonNull String aKey, @NonNull Bitmap aBitmap) {
+        aKey = hashKey(aKey);
         mMemoryCache.put(aKey, aBitmap);
+        String finalAKey = aKey;
```

Comment:
Better assign aKey to finalKey first and use the latter to perform the insertion in the cache.